kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/signal/handlers/DeviceCreationLock.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Areti
+ *******************************************************************************/
+package org.eclipse.kura.nm.signal.handlers;
+
+
+import java.util.Optional;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.kura.nm.NMDbusConnector;
+import org.freedesktop.NetworkManager;
+import org.freedesktop.dbus.connections.impl.DBusConnection;
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.freedesktop.networkmanager.Device;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeviceCreationLock {
+
+    private static final Logger logger = LoggerFactory.getLogger(DeviceCreationLock.class);
+    
+    private final BlockingQueue<Device> deviceQueue;
+    private final NMDeviceCreationHandler handler;
+    private final DBusConnection dbusConnection;
+    
+    public DeviceCreationLock(NMDbusConnector nm, String deviceId) throws DBusException {
+        this.deviceQueue = new ArrayBlockingQueue<Device>(1);
+        this.dbusConnection = nm.getDbusConnection();
+        this.handler = new NMDeviceCreationHandler(nm, deviceId, this.deviceQueue);
+
+        this.dbusConnection.addSigHandler(NetworkManager.DeviceAdded.class, this.handler);
+    }
+
+    public Optional<Device> waitForDeviceCreation() throws DBusException, TimeoutException {
+        try {
+            return Optional.ofNullable(deviceQueue.poll(5L, TimeUnit.SECONDS));
```

Comment:
I would add the timeout as a parameter for testability reasons. Like we did [in the `WPAScanLock`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/signal/handlers/WPAScanLock.java)

```suggestion
    public Optional<Device> waitForDeviceCreation(long deviceCreationTimeoutSeconds) throws DBusException, TimeoutException {
        try {
            return Optional.ofNullable(deviceQueue.poll(deviceCreationTimeoutSeconds, TimeUnit.SECONDS));
```

Additionally, I think a 1 second timeout might be enough and allow for quicker unit tests but I might be wrong. Feel free to correct me on this.